### PR TITLE
Implement Redis for caching fetch requests

### PR DESCRIPTION
**Pull request by jamesjamail**
-pretty straight forward.  Set cached data to expire after 24 hours.
-[link to windows installer for Redis](https://github.com/MicrosoftArchive/redis/releases/download/win-3.0.504/Redis-x64-3.0.504.msi)

### CODE REVIEW

File: wikiHandler.js [hunk 1]
```diff
@@ -86,10 +93,46 @@ function formatPrice(price) {
     }
 }
 
+function parsePage(page, server) {
```
Comment (rm-you) [inline wikiHandler.js:96]: This is duplicated code now -- now that it's a function, can we clean that up below?

File: wikiHandler.js [hunk 2]
```diff
@@ -86,10 +93,46 @@ function formatPrice(price) {
...
+                        parsePage(text, server)
+                    })
+                })
+                .catch(console.error)
+        }
+    })
+
     return fetch(item_url)
         .then(response => response.text())
         .then(text => {
```
Comment (rm-you) [inline wikiHandler.js:138]: Seems like the function parsePage above was pulled from here? So could this be replaced with a call to that somehow? We don't want duplicate codepaths for the same thing if possible.

File: wikiHandler.js [hunk 3]
```diff
@@ -86,10 +93,46 @@ function formatPrice(price) {
...
     // Translate server name to "capitalized" form, eg: GREEN -> Green
     server = server[0] + server.slice(1).toLowerCase();
 
+    //check cache before fetching
+    cache.get(item_url, (err, cached_data) => {
+        if (err) console.error(err);
+        //use cached data if available
+        if (cached_data) {
+            console.log('cached worked for ', item_url)
+            parsePage(cached_data, server)
```
Comment (rm-you) [inline wikiHandler.js:119]: So we're storing the raw page response in cache? I guess this is fine, but it'd be much smaller to cache the actual result we want, right? And then we'd only have to run parsePage once before storing, not every time we fetch.

File: wikiHandler.js [hunk 4]
```diff
@@ -86,10 +93,46 @@ function formatPrice(price) {
...
+                        if (err) console.error(err);
+                        //parse for price
+                        parsePage(text, server)
+                    })
+                })
+                .catch(console.error)
+        }
+    })
+
     return fetch(item_url)
```
Comment (rm-you) [inline wikiHandler.js:136]: Wait, so we fetch here ... but this is also copied somewhere else -- and again, should be a single code-path, this seems duplicated also.

File: wikiHandler.js (hunk 3 above)
Comment (rm-you) [inline wikiHandler.js:119]: Second comment here -- after we do the parsePage, does it ... not return the parsed data? Do we fall through to like 136 below and run fetch() a second time?

Comment (jamesjamail) [reply wikiHandler.js]: 100%.  Was thinking that this morning.  Now it's much more proper.

Comment (jamesjamail) [reply wikiHandler.js]: Oof. Yeah, my bad- remnants of your code i was pulling apart.  Deleted now.

**Pull request merged**